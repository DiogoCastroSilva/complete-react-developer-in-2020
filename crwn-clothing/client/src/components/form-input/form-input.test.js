import React from 'react';
import { shallow } from 'enzyme';

import FormInput from './form-input.component';


describe('Testing FormInput component', () => {
    let wrapper;
    let mockHandleChange

    beforeEach(() => {
        mockHandleChange = jest.fn();

        const mockProps = {
            label: 'email',
            name: 'email',
            handleChange: mockHandleChange
        };

        wrapper = shallow(<FormInput {...mockProps} />);
    });

    it('expects to render the FormInput Component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call handleChanges when input changes', () => {
        wrapper.find('.form-input').simulate('change');

        expect(mockHandleChange).toHaveBeenCalled();
    });

    it('Should not render a label if not passed as prop', () => {
        const mockPropsWithNoLabel = {
            name: 'email',
            handleChange: mockHandleChange
        };

        const noLabelWrapper = shallow(<FormInput {...mockPropsWithNoLabel} />);
        
        expect(noLabelWrapper.exists('.form-input-label')).toBe(false);
    })
});