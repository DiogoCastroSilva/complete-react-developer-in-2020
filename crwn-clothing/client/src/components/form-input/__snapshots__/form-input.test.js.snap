// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FormInput component expects to render the FormInput Component 1`] = `
<styled.div>
  <styled.input
    className="form-input"
    name="email"
    onChange={[MockFunction]}
  />
  <styled.label
    className=" form-input-label"
  >
    email
  </styled.label>
</styled.div>
`;
